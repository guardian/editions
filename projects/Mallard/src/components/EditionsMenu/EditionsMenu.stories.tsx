import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { withKnobs } from '@storybook/addon-knobs'
import { EditionsMenu } from './EditionsMenu'
import { editions, Edition } from 'src/common'
import { SpecialEdition } from '../../../../Apps/common/src'

const props: { specialEditions: SpecialEdition[] } = {
    specialEditions: [
        {
            edition: 'daily-edition' as Edition,
            expiry: new Date(98, 1),
            buttonImageUri:
                'https://media.guim.co.uk/49cebb0db4a3e4d26d7d190da7be4a2e9bd7534f/0_0_103_158/103.png',
            topic: 'food',
            editionType: 'Special',
            notificationUTCOffset: 0,
            header: {
                title: `Food
Monthly`,
            },
            title: `Food
Monthly`,
            subTitle:
                'Store cupboard special: 20 quick and easy lockdown suppers',
            buttonStyle: {
                backgroundColor: '#FEEEF7',
                expiry: {
                    color: '#7D0068',
                    font: 'GuardianTextSans-Regular',
                    lineHeight: 16,
                    size: 15,
                },

                subTitle: {
                    color: '#7D0068',
                    font: 'GuardianTextSans-Bold',
                    lineHeight: 20,
                    size: 17,
                },
                title: {
                    color: '#121212',
                    font: 'GHGuardianHeadline-Regular',
                    lineHeight: 34,
                    size: 34,
                },
                image: {
                    height: 134,
                    width: 87,
                },
            },
        },
    ],
}

storiesOf('EditionsMenu', module)
    .addDecorator(withKnobs)
    .add('EditionsMenu - default', () => (
        <EditionsMenu
            navigationPress={() => {}}
            selectedEdition={editions.daily}
            storeSelectedEdition={() => {}}
        />
    ))
    .add('EditionsMenu - with Special Edition', () => (
        <EditionsMenu
            navigationPress={() => {}}
            selectedEdition={editions.daily}
            storeSelectedEdition={() => {}}
            {...props}
        />
    ))
