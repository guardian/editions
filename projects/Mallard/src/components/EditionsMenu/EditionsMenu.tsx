import React from 'react'
import { FlatList, ScrollView } from 'react-native'
import { EditionId, RegionalEdition, SpecialEdition } from 'src/common'
import { StoreSelectedEditionFunc } from 'src/hooks/use-edition-provider'
import { defaultRegionalEditions } from '../../../../Apps/common/src/editions-defaults'
import { EditionsMenuHeader } from './Header/Header'
import { ItemSeperator } from './ItemSeperator/ItemSeperator'
import { RegionButton } from './RegionButton/RegionButton'
import { SpecialEditionButton } from './SpecialEditionButton/SpecialEditionButton'

const EditionsMenu = ({
    navigationPress,
    regionalEditions,
    selectedEdition,
    specialEditions,
    storeSelectedEdition,
}: {
    navigationPress: () => void
    regionalEditions?: RegionalEdition[]
    selectedEdition: EditionId
    specialEditions?: SpecialEdition[]
    storeSelectedEdition: StoreSelectedEditionFunc
}) => {
    return (
        <ScrollView>
            <EditionsMenuHeader>Regions</EditionsMenuHeader>
            <FlatList
                data={regionalEditions || defaultRegionalEditions}
                renderItem={({ item }: { item: RegionalEdition }) => {
                    return (
                        <RegionButton
                            selected={
                                selectedEdition === item.edition ? true : false
                            }
                            onPress={() => {
                                storeSelectedEdition(item, 'RegionalEdition')
                                navigationPress()
                            }}
                            title={item.title}
                            subTitle={item.subTitle}
                        />
                    )
                }}
                ItemSeparatorComponent={() => <ItemSeperator />}
            />
            {specialEditions && specialEditions.length > 0 && (
                <>
                    <EditionsMenuHeader>Special Editions</EditionsMenuHeader>
                    <FlatList
                        data={specialEditions}
                        renderItem={({ item }: { item: SpecialEdition }) => {
                            const {
                                buttonStyle,
                                buttonImageUri,
                                edition,
                                expiry,
                                title,
                                subTitle,
                            } = item
                            return (
                                <SpecialEditionButton
                                    buttonImageUri={buttonImageUri}
                                    expiry={expiry}
                                    onPress={() => {
                                        storeSelectedEdition(
                                            item,
                                            'SpecialEdition',
                                        )
                                        navigationPress()
                                    }}
                                    title={title}
                                    selected={
                                        selectedEdition === edition
                                            ? true
                                            : false
                                    }
                                    style={buttonStyle}
                                    subTitle={subTitle}
                                />
                            )
                        }}
                    />
                </>
            )}
        </ScrollView>
    )
}

export { EditionsMenu }
