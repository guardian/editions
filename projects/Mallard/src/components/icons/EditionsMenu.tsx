import React from 'react';
import Svg, { Circle } from 'react-native-svg';

const EditionsMenu = () => (
	<Svg width="40" height="40" viewBox="0 0 42 42" fill="none">
		<Circle cx="21" cy="21" r="20.5" fill="#052962" stroke="white" />
		<Circle cx="14.1751" cy="13.125" r="6.825" fill="white" />
		<Circle cx="14.1751" cy="27.825" r="6.825" fill="#FF7F0F" />
		<Circle cx="28.875" cy="13.125" r="6.825" fill="#FFBAC8" />
		<Circle cx="28.875" cy="27.825" r="6.825" fill="#90DCFF" />
	</Svg>
);

export { EditionsMenu };
