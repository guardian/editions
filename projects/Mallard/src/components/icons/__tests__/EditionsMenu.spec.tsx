import React from 'react';
import TestRenderer from 'react-test-renderer';
import { EditionsMenu } from '../EditionsMenu';

describe('EditionsMenu', () => {
	it('should display a EditionsMenu icon in SVG', () => {
		const component = TestRenderer.create(<EditionsMenu />).toJSON();
		expect(component).toMatchSnapshot();
	});
});
