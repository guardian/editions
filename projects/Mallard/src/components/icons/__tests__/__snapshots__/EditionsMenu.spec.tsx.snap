// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditionsMenu should display an EditionsMenu icon in SVG 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="40"
  bbWidth="40"
  fill="none"
  focusable={false}
  height="40"
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 40,
        "width": 40,
      },
    ]
  }
  vbHeight={42}
  vbWidth={42}
  width="40"
>
  <RNSVGGroup
    fill={null}
    propList={
      Array [
        "fill",
      ]
    }
  >
    <RNSVGCircle
      cx="21"
      cy="21"
      fill={
        Object {
          "payload": 4278528354,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
          "stroke",
        ]
      }
      r="20.5"
      stroke={
        Object {
          "payload": 4294967295,
          "type": 0,
        }
      }
    />
    <RNSVGCircle
      cx="14.1751"
      cy="13.125"
      fill={
        Object {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="6.825"
    />
    <RNSVGCircle
      cx="14.1751"
      cy="27.825"
      fill={
        Object {
          "payload": 4294934287,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="6.825"
    />
    <RNSVGCircle
      cx="28.875"
      cy="13.125"
      fill={
        Object {
          "payload": 4294949576,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="6.825"
    />
    <RNSVGCircle
      cx="28.875"
      cy="27.825"
      fill={
        Object {
          "payload": 4287683839,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="6.825"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
